lib/repl.js

```diff
@@ -691,11 +691,10 @@ REPLServer.prototype.setPrompt = function setPrompt(prompt) {
   REPLServer.super_.prototype.setPrompt.call(this, prompt);
 };
 
-REPLServer.prototype.turnOffEditorMode = function() {
-  this.editorMode = false;
-  this.setPrompt(this._initialPrompt);
-};
-
+REPLServer.prototype.turnOffEditorMode = util.deprecate(
+  _turnOffEditorMode,
```

Comment:
I think it would be nicer to have `_turnOffEditorMode(this);` (or self) instead of `_turnOffEditorMode.call(this);`
It is weird that `turnOn` and `turnOff` works differently.

In that case the prototype function would have to be changed to look somewhat like
```js
function () {
  _turnOffEditorMode(this);
}
```